templates/subscription-centre/index.html

```diff
@@ -84,132 +77,33 @@ <h2 class="p-modal__title" id="modal-title">Unsubscribe</h2>
   </section>
 </div>
 
+<script src="/static/js/src/modal.js"></script>
 <script>
-  (function () {
-  var currentDialog = null;
-  var lastFocus = null;
-  var ignoreFocusChanges = false;
-  var focusAfterClose = null;
-  const unsubscribeButton =  document.querySelector("#unsubscribe");
-
-  // Traps the focus within the currently open modal dialog
-  function trapFocus(e) {
-    if (ignoreFocusChanges) return;
-
-    if (currentDialog.contains(e.target)) {
-      lastFocus = e.target;
-    } else {
-      focusFirstDescendant(currentDialog);
-      if (lastFocus == document.activeElement) {
-        focusLastDescendant(currentDialog);
-      }
-      lastFocus = document.activeElement;
-    }
-  }
-
-  // Attempts to focus given element
-  function attemptFocus(child) {
-    if (child.focus) {
-      ignoreFocusChanges = true;
-      child.focus();
-      ignoreFocusChanges = false;
-      return document.activeElement === child;
-    }
-
-    return false;
-  }
-
-  // Focuses first child element
-  function focusFirstDescendant(element) {
-    for (var i = 0; i < element.childNodes.length; i++) {
-      var child = element.childNodes[i];
-      if (attemptFocus(child) || focusFirstDescendant(child)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  // Focuses last child element
-  function focusLastDescendant(element) {
-    for (var i = element.childNodes.length - 1; i >= 0; i--) {
-      var child = element.childNodes[i];
-      if (attemptFocus(child) || focusLastDescendant(child)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  /**
-    Toggles visibility of modal dialog.
-    @param {HTMLElement} modal Modal dialog to show or hide.
-    @param {HTMLElement} sourceEl Element that triggered toggling modal
-    @param {Boolean} open If defined as `true` modal will be opened, if `false` modal will be closed, undefined toggles current visibility.
-  */
-  function toggleModal(modal, sourceEl, open) {
-    if (modal && modal.classList.contains('p-modal')) {
-      if (typeof open === 'undefined') {
-        open = modal.style.display === 'none';
-      }
-
-      if (open) {
-        currentDialog = modal;
-        modal.style.display = 'flex';
-        focusFirstDescendant(modal);
-        focusAfterClose = sourceEl;
-        document.addEventListener('focus', trapFocus, true);
-      } else {
-        modal.style.display = 'none';
-        if (focusAfterClose && focusAfterClose.focus) {
-          focusAfterClose.focus();
-        }
-        document.removeEventListener('focus', trapFocus, true);
-        currentDialog = null;
-      }
-    }
+  initModals('#modal', '[aria-controls=modal]', false)
```

Comment:
You may want to wrap this in a `(function() {})()`? If page takes time to load or one of the HTML elements doesn't load as fast as the others it may fail to initialise. That will make sure it loads when the page is ready (reaches that function and calls itself)